cmd/deployment/search.go

```diff
@@ -51,21 +54,57 @@ var searchCmd = &cobra.Command{
 			return err
 		}
 
-		res, err := deploymentapi.Search(deploymentapi.SearchParams{
-			API:     ecctl.Get().API,
-			Request: &sr,
-		})
+		returnAllMatches, _ := strconv.ParseBool(cmd.Flag("all-matches").Value.String())
+		if returnAllMatches && sr.Sort == nil {
+			return fmt.Errorf("The query must include a sort-field when using --all-matches. Example: \"sort\": [\"id\"]")
+		}
 
-		if err != nil {
-			return err
+		var result *models.DeploymentsSearchResponse
+		var cursor string
+		for {
+			sr.Cursor = cursor
+			if returnAllMatches {
+				// Custom batch-size to override any size already set in the input query
+				sr.Size = 500
```

Comment:
Does it make sense to read it from cmd arg as well? (while "500" can be the default value).

Reply:
Yeah it could be useful if this size turns out to be wrong in some situations

Reply:
I added a --size parameter